deals/module/retrieve.go

```diff
@@ -103,6 +103,9 @@ func (m *Module) retrieve(ctx context.Context, lapi *apistruct.FullNodeStruct, l
 	go func() {
 		defer lapiCls()
 		defer close(out)
+		m.metricRetrievalTracking.Add(ctx, 1)
+		defer m.metricRetrievalTracking.Add(ctx, -1)
+
```

Comment:
Yeah, that could be possible. I'd vote to let the daemon itself publish the metric instead of being an external component, just at to make metrics a baked-in feature of the daemon.